scenario/constraints/TokenBalanceConstraint.ts

```diff
@@ -29,14 +29,22 @@ export class TokenBalanceConstraint<T extends CometContext, R extends Requiremen
       const solutions = [];
       solutions.push(async function barelyMeet(context: T, world: World) {
         for (const assetName in actorsByAsset) {
-          const asset = await getAssetFromName(assetName, context)
+          const asset = await getAssetFromName(assetName, context);
           for (const actorName in actorsByAsset[assetName]) {
+            const cometActor = context.actors[actorName];
             const actor = await getActorAddressFromName(actorName, context);
             const amount: ComparativeAmount = actorsByAsset[assetName][actorName];
             const balance = await asset.balanceOf(actor);
             const decimals = await asset.token.decimals();
             const toTransfer = getToTransferAmount(amount, balance, decimals);
-            await context.sourceTokens(world, toTransfer, asset.address, actor);
+            if (toTransfer < 0) {
+              // Send extra balance to the other address
+              const leftoverHolderAddress = '0x0000000000000000000000000000000000000001';
```

Comment:
Maybe a const at the top of the file?

Reply:
BTW, there *is* `removeTokens` in TokenSourcer, should we be triggering that?

Reply:
Yeah, TokenSourcer *should* be handling negative transfers, so maybe we could move that fix there.

Here's the code in TokenSourcer for reference (not sure why it's not working properly): https://github.com/compound-finance/comet/blob/main/plugins/scenario/utils/TokenSourcer.ts#L26-L27